### PR TITLE
deal with null entities in AuthorizationFramework

### PR DESCRIPTION
**Pull request by vladak**
This provides basic protection for `null` entities.

### CODE REVIEW

File: opengrok-indexer/src/main/java/org/opengrok/indexer/authorization/AuthorizationFramework.java [hunk 1]
```diff
@@ -763,6 +763,11 @@ private boolean checkAll(HttpServletRequest request, String cache, Nameable enti
             return true;
         }
 
+        if (entity == null) {
+            LOGGER.log(Level.WARNING, "entity was null for request {}", request);
```
Comment (tulinkry) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/authorization/AuthorizationFramework.java]: maybe an entity class (getSimpleName) would also be a benefit for debuging

Comment (tulinkry) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/authorization/AuthorizationFramework.java]: and did you try what request.toString returns? I guess it's not any valueable info just the reference number.

Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/authorization/AuthorizationFramework.java]: will use `request.getRequestURI()`

Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/authorization/AuthorizationFramework.java]: or maybe `request.getParameterMap()` would be more useful.

Comment (tulinkry) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/authorization/AuthorizationFramework.java]: may be. a little bit too long eventually for searches

Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/authorization/AuthorizationFramework.java]: yes, however might be crucial for figuring out what went wrong.

**Pull request merged**